extensions/basic_auth/plugin.cc

```diff
@@ -0,0 +1,330 @@
+    return false;
+  }
+  // j is a JsonObject holds configuration data
+  auto j = result.value();
+  // basic_auth_configuration_ container has the following example structure
+  //{
+  // "exact":{
+  //    "/products/api/store":{
+  //      ["GET"],
+  //      ["YWRtaW46YWRtaW4="]
+  //    }
+  // },
+  // "prefix":{
+  //    "/reviews":{
+  //      ["GET", "POST"],
+  //      ["YWRtaW46YWRtaW4=", "AWRtaW46YWRtaW4="]
+  //    }
+  //  },
+  // "suffix":{
+  //    "/api/pay":{
+  //      ["GET", "POST"],
+  //      ["YWRtaW46YWRtaW4=", "AWRtaW46YWRtaW4="]
+  //    }
+  //  }
+  //}
+  if (!JsonArrayIterate(
+          j, "basic_auth_rules", [&](const json& configuration) -> bool {
+            auto request_path =
+                JsonGetField<std::string>(configuration, "request_path")
+                    .value_or("");
+            auto match =
+                JsonGetField<std::string>(configuration, "match").value_or("");
+            if (request_path == "" || match == "") {
+              return false;
+            }
+            struct headerData header;
+            if (!JsonArrayIterate(
+                    configuration, "request_methods",
+                    [&](const json& methods) -> bool {
+                      auto method = JsonValueAs<std::string>(methods);
+                      if (method.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        LOG_WARN("unexpected method");
+                        return false;
+                      }
+                      header.request_methods.insert(
+                          std::string(method.first.value()));
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request methods.");
+              return false;
+            }
+
+            if (!JsonArrayIterate(
+                    configuration, "credentials",
+                    [&](const json& credentials) -> bool {
+                      auto credential = JsonValueAs<std::string>(credentials);
+                      if (credential.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        LOG_WARN("unexpected credential");
+                        return false;
+                      }
+                      header.encoded_credentials.insert(
+                          Base64::encode(credential.first.value().data(),
+                                         credential.first.value().size()));
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for credentials.");
+              return false;
+            }
+            auto path_iter = basic_auth_configuration_.find(match);
+            if (path_iter != basic_auth_configuration_.end() &&
+                path_iter->second.find(request_path) !=
+                    basic_auth_configuration_[match].end()) {
```

Comment:
This looks like a second lookup. You can just path_iter->second.end().

Reply:
okok that's true!